L1Trigger/TrackFindingTracklet/src/VMRouterCM.cc

```diff
@@ -262,22 +358,59 @@ void VMRouterCM::execute(unsigned int) {
           vmstubsMEPHI_[i]->addStub(vmstub, ivm * nvmmebins_ + vmbin);
       }
 
-      //Fill the TE VM memories
-      if (layerdisk_ >= N_LAYER && (!stub->isPSmodule()))
-        continue;
-
       for (auto& ivmstubTEPHI : vmstubsTEPHI_) {
         unsigned int iseed = ivmstubTEPHI.seednumber;
-        unsigned int lutwidth = settings_.lutwidthtab(1, iseed);
+
+        // This flag is used to replicate the behavior of the old VMRouter for
+        // the case of the triplet seeds.
+        const bool isTripletSeed = (iseed >= L2L3L4);
+
+        if (!isTripletSeed && layerdisk_ >= N_LAYER && (!stub->isPSmodule()))
+          continue;
+        unsigned int inner = (!isTripletSeed ? 1 : ivmstubTEPHI.stubposition);
+        unsigned int lutwidth = settings_.lutwidthtab(inner, iseed);
+        if (settings_.extended()) {
+          lutwidth = settings_.lutwidthtabextended(inner, iseed);
+        }
 
         int lutval = -999;
 
-        if (layerdisk_ < N_LAYER) {
-          lutval = melut;
+        if (inner > 0) {
+          if (layerdisk_ < N_LAYER) {
+            lutval = (!isTripletSeed ? melut : melutOld);
+          } else {
+            if (inner == 2 && iseed == Seed::L2L3D1) {
+              lutval = 0;
+              if (stub->r().value() < 10) {
+                lutval = 8 * (1 + (stub->r().value() >> 2));
+              } else {
+                if (stub->r().value() < settings_.rmindiskl3overlapvm() / settings_.kr()) {
+                  lutval = -1;
+                }
+              }
+            } else {
+              lutval = (!isTripletSeed ? diskTable_.lookup((indexz << nbitsrfinebintable_) + indexr)
+                                       : diskTableOld_.lookup((indexzOld << nbitsrfinebintable_) + indexrOld));
+              if (lutval == 0)
+                continue;
+            }
+          }
+          if (lutval == -1)
+            continue;
         } else {
-          lutval = diskTable_.lookup((indexz << nbitsrfinebintable_) + indexr);
-          if (lutval == 0) {
+          if (iseed < Seed::L1D1 || iseed > Seed::L2D1) {
+            lutval = innerTable_.lookup((indexzOld << nbitsrfinebintable_) + indexrOld);
+          } else {
+            lutval = innerOverlapTable_.lookup((indexzOld << nbitsrfinebintable_) + indexrOld);
+          }
+          if (lutval == -1)
             continue;
+          if (settings_.extended() &&
+              (iseed == Seed::L3L4 || iseed == Seed::L5L6 || iseed == Seed::D1D2 || iseed == Seed::L2L3D1)) {
+            int lutval2 = innerThirdTable_.lookup((indexzOld << nbitsrfinebintable_) + indexrOld);
+            if (lutval2 != -1) {
+              lutval += (lutval2 << 10);
```

Comment:
Magic numbers, such as 10 are forbidden. All such numbers must be replaced by named constants.